### PR TITLE
SizeObserver crusade: drawer

### PR DESCRIPTION
**Pull request by Hixie**
Drawer doesn't need a SizeObserver, since it only looks at the size
in event handlers. It can just go and probe the tree to read the size.

Also, change from using _kEdgeDragWidth to using _kWidth when figuring
out how much of the drawer to show when dragging it from the edge, since
that is more likely to match the drawer's width.

### CODE REVIEW

File: packages/flutter/lib/src/material/drawer.dart [hunk 1]
```diff
@@ -116,19 +116,19 @@ class DrawerControllerState extends State<DrawerController> {
...
-    });
-  }
 
   void _handleTapDown(Point position) {
     _controller.stop();
     _ensureHistoryEntry();
   }
 
+  double get _width {
+    RenderBox drawerBox = _drawerKey.currentContext?.findRenderObject();
```
Comment (abarth) [inline packages/flutter/lib/src/material/drawer.dart]: Maybe add an `assert(debugDispatchingEvents)`

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/material/drawer.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/material/drawer.dart]: There's too many entry points for "events". Touch events, keyboard events, semantics events, etc. But they all happen between frames. So I went with:
```dart
assert(!Scheduler.debugInFrame);
```

**Comment by abarth** [comment]
ok

**Pull request merged**